arangod/Aql/EnumerateCollectionExecutor.cpp

```diff
@@ -48,22 +49,20 @@
 using namespace arangodb;
 using namespace arangodb::aql;
 
-namespace {
-std::vector<size_t> const emptyAttributePositions;
-}
-
 EnumerateCollectionExecutorInfos::EnumerateCollectionExecutorInfos(
     RegisterId outputRegister, aql::QueryContext& query,
     Collection const* collection, Variable const* outVariable,
     bool produceResult, Expression* filter,
-    arangodb::aql::Projections projections, bool random, bool count,
-    ReadOwnWrites readOwnWrites)
+    arangodb::aql::Projections projections,
```

Comment:
Yes, this is intentional. Projections are owned by the EnumerationCollectionNode (via DocumentProducingNode), and we don't want to steal the projections from the node. So we have to do one copy. The copy we take is then moved into the local `_projections` instance variable in `EnumerateCollectionExecutorInfos`. We can also accept the projections by const reference and then copy. But we will still have to copy them once, because I don't want to steal/alias existing projections from the node.